packages/fbjs/src/core/shallowEqual.js

```diff
@@ -25,7 +25,7 @@ function is(x: mixed, y: mixed): boolean {
   // SameValue algorithm
   if (x === y) { // Steps 1-5, 7-10
     // Steps 6.b-6.e: +0 != -0
-    return x !== 0 || 1 / (x: $FlowIssue) === 1 / (y: $FlowIssue);
+    return x !== 0 || y !== 0 || 1 / x === 1 / y;
```

Comment:
I know this extra inequality check is redundant, but it makes Flow happy. At the moment, Flow doesn't understand that, if `x === y`, then refining `x` also refines `y`.